### PR TITLE
CoreAPI and CitizenAPI

### PR DESCRIPTION
**Pull request by Monika-Mistry**
Core and Citizen API coded and tested. Docker  and docker-compose set up and tested.

### CODE REVIEW

File: CitizenAPI/src/main/java/com/bae/citizen/repository/CitizenRepository.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.bae.citizen.repository;
+
+import java.util.List;
+
+import org.springframework.data.repository.CrudRepository;
+
+import com.bae.citizen.domain.Citizen;
+
```
Comment (JHarry444) [inline CitizenAPI/src/main/java/com/bae/citizen/repository/CitizenRepository.java]: @Repository

File: CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.bae.citizen.domain.Citizen;
+import com.bae.citizen.service.CitizenService;
+
+@RestController
+public class CitizenController {
+
+	private CitizenService citizenService;
+
+	public CitizenController() {
```
Comment (JHarry444) [inline CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java]: Don't think you want a default constructor - if there's no service this class will fall over.

File: CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+	}
+
+	@Autowired
+	public CitizenController(CitizenService citizenService) {
+
+		this.citizenService = citizenService;
+	}
+
+	@GetMapping("/getCitizensByForenames/{username}/{forenames}")
```
Comment (JHarry444) [inline CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java]: What purpose does the '/username' server?

File: CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	}
+
+	@GetMapping("/getCitizensBySurname/{username}/{surname}")
+	public List<Citizen> findCitizensBySurname(@PathVariable("surname") String surname) {
+
+		return citizenService.findBySurname(surname);
+	}
+
+	@GetMapping("/getCitizensByFullname/{username}/{forenames}/{surname}")
+	public List<Citizen> findCitizensByFullname(@PathVariable("forenames") String forenames,
```
Comment (JHarry444) [inline CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java]: Might be better with ResponseEntity.

File: CitizenAPI/src/main/java/com/bae/citizen/service/CitizenServiceImplementation.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import com.bae.citizen.domain.Citizen;
+import com.bae.citizen.repository.CitizenRepository;
+
+@Service
+public class CitizenServiceImplementation implements CitizenService{
+
+	
+	private CitizenRepository citizenRepository;
+	
+	public CitizenServiceImplementation() {
```
Comment (JHarry444) [inline CitizenAPI/src/main/java/com/bae/citizen/service/CitizenServiceImplementation.java]: Remove empty constructor.

File: CitizenAPI/src/test/java/com/bae/citizen/rest/CitizenMockitoTests.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+		List<Citizen> MOCK_LIST = new ArrayList<>();
+		MOCK_LIST.add(TestConstants.MOCK_CITIZEN_1);
+		MOCK_LIST.add(TestConstants.MOCK_CITIZEN_2);
+
+		List<Citizen> MOCK_RETURN_LIST = new ArrayList<>();
+		MOCK_RETURN_LIST.add(TestConstants.MOCK_CITIZEN_1);
+
+		Mockito.when(service.findBySurname("Aarvark")).thenReturn(MOCK_RETURN_LIST);
+
+		assertEquals(MOCK_RETURN_LIST, controller.findCitizensBySurname("Aarvark"));
```
Comment (JHarry444) [inline CitizenAPI/src/test/java/com/bae/citizen/rest/CitizenMockitoTests.java]: redundant

File: CoreAPI/src/main/java/com/bae/Constants.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.bae;
+
+public class Constants {
+	
+	public static final String CITIZEN_URL = "http://citizen:8003/";
+	public static final String CITIZEN_FORENAME_URL = "http://citizen:8003/getCitizensByForenames/";
```
Comment (JHarry444) [inline CoreAPI/src/main/java/com/bae/Constants.java]: Could be CITIZEN_URL + "/getCitizensByForenames/"?

File: CoreAPI/src/main/java/com/bae/service/CoreServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.bae.domain.User;
+
+@Service
+public class CoreServiceImpl implements CoreService{
+
+
+	public boolean verifyLogin(User user) {
+		boolean condition =  Strings.isNullOrEmpty(user.getPassword()) || Strings.isNullOrEmpty(user.getUsername());
+		return condition ? false : true;
```
Comment (JHarry444) [inline CoreAPI/src/main/java/com/bae/service/CoreServiceImpl.java]: return !condition;

File: CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java (hunk 3 above)
Comment (Monika-Mistry) [reply CitizenAPI/src/main/java/com/bae/citizen/rest/CitizenController.java]: Match the endpoint in core. We only want to allow access to logged in users.

**Comment by Monika-Mistry** [comment]
Restructuring

**Pull request closed without merging**